### PR TITLE
feat: support for did element

### LINKED ISSUE #138: Interop Test: DID Element

### PR DESCRIPTION
**Pull request by sudeshrshetty**
- addes support for did element and tests
- closes #138

Signed-off-by: sudesh.shetty <sudesh.shetty@securekey.com>

### CODE REVIEW

**Comment by fqutishat** [comment]
@sudeshrshetty  after this PR is merged we need to add new profile in sandbox https://github.com/trustbloc/edge-sandbox/blob/master/test/bdd/fixtures/scripts/vcs_configure.sh#L42

https://github.com/trustbloc/edge-sandbox/blob/master/cmd/issuer-rest/static/index.html#L39

**Comment by sudeshrshetty** [comment]
> @sudeshrshetty after this PR is merged we need to add new profile in sandbox https://github.com/trustbloc/edge-sandbox/blob/master/test/bdd/fixtures/scripts/vcs_configure.sh#L42
> 
> https://github.com/trustbloc/edge-sandbox/blob/master/cmd/issuer-rest/static/index.html#L39

Sure

File: pkg/restapi/vc/operation/operations.go [hunk 1]
```diff
@@ -990,16 +999,29 @@ func (o *Operation) parseAndVerifyVP(vpBytes []byte) (*verifiable.Presentation,
...
+		var publicKeyID string
+
+		for _, k := range didDoc.PublicKey {
+			if strings.HasPrefix(k.Type, Ed25519VerificationKey) {
+				publicKeyID = k.ID
+				break
+			}
+		}
+
+		if !isDID(publicKeyID) {
```
Comment (sandrask) [inline pkg/restapi/vc/operation/operations.go]: So their DID keys are not URIs (not as per spec). Can you add comment (or create issue) to remove this condition when they become compliant?

Comment (sudeshrshetty) [reply pkg/restapi/vc/operation/operations.go]: DID element has existing issue to support public keys in with DID based ids. Created #140 
 and added a TODO to remove it in future.

**Comment by sudeshrshetty** [comment]
> I'm not a golang expert, but the JS parts look correct. Great work!

Thanks @OR13

File: pkg/restapi/vc/operation/operations.go (hunk 1 above)
Comment (OR13) [reply pkg/restapi/vc/operation/operations.go]: They should be compliant now: https://github.com/decentralized-identity/element/pull/179

https://uniresolver.io/1.0/identifiers/did:elem:EiBJJPdo-ONF0jxqt8mZYEj9Z7FbdC87m2xvN0_HAbcoEg

**Pull request merged**